cmake/compiler.cmake

```diff
@@ -211,6 +211,13 @@ option(ENABLE_WERROR "Make all compiler warnings into errors" OFF)
 
 option(ENABLE_UB_SANITIZER "Make the compiler generate runtime code to perform undefined behaviour checks" OFF)
 
+option(ENABLE_HARDENING "Enable compiler options that harden against memory corruption attacks" ON)
+if (ENABLE_HARDENING)
+    set(HARDENING_FLAGS "-fstack-protector")
+    add_compile_flags("C;CXX" ${HARDENING_FLAGS})
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
TBH, I don't understand a principle:

* The goal is to add PIC/PIE to our own object files (and support ASLR for the executable)?
* Or to enable PIC/PIE for all code that is compiled into the executable?

If it is the latter, it seems, at least some libraries are missed: ones that are enabled using `ExternalProject_Add()`. Will not better to enable `-fPIC` everywhere (and possibly miss some optimizations) rather than don't enable neither `-fPIC`, nor `-fPIE`? If some code is build without PIC/PIE, will ASLR actually enabled (and how to check it)?

I guess it worth to discuss our goals and ways to achieve them.

Reply:
The latter, so `-fPIC` is passed explicitly via CFLAGS/CXXFLAGS.